codegenerator/cli/templates/dynamic/codegen/src/Types.res.hbs

```diff
@@ -229,44 +229,11 @@ module {{contract.name.capitalized}} = {
     let name = "{{event.name.capitalized}}"
     let contractName = "{{contract.name.capitalized}}"
     let topic0 = "{{event.topic0}}"
-
-    {{#if event.params}}
-    @genType
-    type eventArgs = {
-      {{#each event.params as | param |}}
-      @as("{{param.js_name}}")
-      {{param.res_name}}: {{param.res_type}},
-      {{/each}}
-    }
-
-    let eventArgsSchema = S.object(s => {
-    {{#each event.params as | param |}}
-      {{param.res_name}}: s.field("{{param.res_name}}", {{param.res_schema_code}}),
-    {{/each}}
-    })
-
-    let convertHyperSyncEventArgs = (decodedEvent: HyperSyncClient.Decoder.decodedEvent): eventArgs => {
-      {
-        {{#each event.indexed_params as | param index |}}
-        {{param.res_name}}: decodedEvent.indexed->Js.Array2.unsafe_get({{index}})->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
-        {{/each}}
-        {{#each event.body_params as | param index |}}
-        {{param.res_name}}: decodedEvent.body->Js.Array2.unsafe_get({{index}})->HyperSyncClient.Decoder.toUnderlying->Utils.magic,
-        {{/each}}
-      }
-    }
-
-    let decodeHyperFuelData = (_) => Js.Exn.raiseError("HyperFuel decoder not implemented")
-    {{else}}
     @genType
-    type eventArgs = unit
-
-    let eventArgsSchema = S.literal(%raw(`null`))->S.variant((. _) => ())
-
-    external convertHyperSyncEventArgs: HyperSyncClient.Decoder.decodedEvent => eventArgs = "%identity"
```

Comment:
The only regression is that it's not runtime-free anymore, but everything else should work the same as before.

Reply:
Doesn't the Utils.magic function become zero cost during optimization steps?

Reply:
Only when the function is called from the same file